### ISSUE TITLE
Eager Few Shot OD with EfficientDet D2

### ISSUE THREAD

**Issue by colabwarrior** [tensorflow/models#9328]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x ] I am using the latest TensorFlow Model Garden release and TensorFlow 2.
- [ x] I am reporting the issue to the correct repository. (Model Garden official or research directory)
- [ x] I checked to make sure that this issue has not already been filed.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/blob/master/research/object_detection/colab_tutorials/eager_few_shot_od_training_tf2_colab.ipynb

## 2. Describe the bug

RetinaNet does a good job detecting the custom object when used in eager few shot learning code. I tried replacing RetinaNet with EfficientDet D2. Made the changes to shape i.e  768x768 where required. The model seems to be training but does not detect the custom object. I checked the checkpoint file and seems like no changes are required to load the model from checkpoint and also the "to_fine_tune" weights have the same name. Only thing that was required to be changed was the shape from 640x640 to 768x768 which I did. But EfficientDet D2 does not seem to detect anything.

Any help is appreciated on how to use EfficientDet models in eager few shot training mode.

## 6. System information

Running on google Colab Pro.

**Comment by ravikyram** [comment]
@colabwarrior 

I have tried in TF nightly version. Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/166dffc3257a660afe201651a2c94ec2/untitled423.ipynb#scrollTo=1V8OAB1wQAEV). You are also seeing the same behavior?
Thanks!

**Comment by colabwarrior** [comment]
Thanks for getting back on this. Yes, I do see the same behavior i.e. No bounding boxes around the object of interest when I use EfficientDet models.

**Comment by DanilKonon** [comment]
@colabwarrior how did you manage to load efficientdet model for fine-tuning? There is RetinaNet in an example.

**Comment by colabwarrior** [comment]
Please see the file link below for EfficientDet. It shows how I loaded EfficientDet model but after training it does not detect my custom object i.e. No bounding boxes.  Thanks for your help.

https://colab.research.google.com/drive/1EixYblennyoYhMgjr5Rcba_eooW89a0h?usp=sharing

**Comment by DanilKonon** [comment]
Yes, I have the same problem. I don't think you can just replace model path here, you need to replace variables in checkpoint too.

```
fake_box_predictor=tf.compat.v2.train.Checkpoint(
    _base_tower_layers_for_heads=detection_model._box_predictor._base_tower_layers_for_heads,
    _box_prediction_head=detection_model._box_predictor._box_prediction_head)

fake_model=tf.compat.v2.train.Checkpoint(_feature_extractor=detection_model._feature_extractor,
                                         _box_predictor=fake_box_predictor)
```
But I don't know what to write instead of  _base_tower_layers_for_heads and so on. I wrote a [stackoverflow question](https://stackoverflow.com/questions/64277009/eager-few-shot-object-detection-colab-for-centernet) but no-one answered it yet...

**Comment by colabwarrior** [comment]
I checked the checkpoint file (for efficientdet models) for variable names using the command below and the variables have the same name. That's why I loaded it the same way. I am not sure why it does not work.

                   tf.train.list_variables(tf.train.latest_checkpoint('./tf_ckpts/'))

**Comment by DanilKonon** [comment]
Ok, thank you! Really great function. I printed some layers, but I don't think there are any _box_predictor._box_prediction_head layers in EfficientDet. Where should I look for it?

```
'model/_box_predictor/_base_tower_layers_for_heads/box_encodings',
 'model/_box_predictor/_base_tower_layers_for_heads/class_predictions_with_background',
 'model/_box_predictor/_box_prediction_head/_box_encoder_layers',
 'model/_box_predictor/_head_scope_conv_layers/BoxPredictionTower',
 'model/_box_predictor/_head_scope_conv_layers/ClassPredictionTower',
 'model/_box_predictor/_prediction_heads/class_predictions_with_background'
```

**Comment by colabwarrior** [comment]
'_box_predictor._box_prediction_head' layers are there in the checkpoint file for efficientdet. I have checked it. Also it is in the output you pasted in your comment. 

I am not sure about the answer to your second question but the same prefixes exists in both models (RetinaNet and EfficientDet)
